src/uct/tcp/tcp_cm.c

```diff
@@ -75,200 +82,426 @@ static ucs_status_t uct_tcp_cm_io_err_handler_cb(void *arg, int io_errno)
     return UCS_ERR_NO_PROGRESS;
 }
 
-static void uct_tcp_cm_trace_conn_pkt(const uct_tcp_ep_t *ep, const char *msg,
-                                      const struct sockaddr_in *peer_addr)
+static void uct_tcp_cm_event_str(uct_tcp_cm_conn_event_t event,
+                                 char *event_str, size_t max_len)
 {
-    char str_addr[UCS_SOCKADDR_STRING_LEN];
+    char *p = event_str;
 
-    ucs_debug("tcp_ep %p: %s %s", ep, msg,
-              ucs_sockaddr_str((const struct sockaddr*)peer_addr,
-                               str_addr, UCS_SOCKADDR_STRING_LEN));
+    if (event & UCT_TCP_CM_CONN_REQ) {
+        ucs_snprintf_zero(event_str, max_len, "%s",
+                          UCS_PP_MAKE_STRING(UCT_TCP_CM_CONN_REQ));
+        p       += strlen(event_str);
+        max_len -= strlen(event_str);
+    }
+
+    if (event & UCT_TCP_CM_CONN_ACK) {
+        if (p != event_str) {
+            ucs_snprintf_zero(p, max_len, " | ");
+            p       += strlen(event_str);
+            max_len -= strlen(event_str);
+        }
+        ucs_snprintf_zero(p, max_len, "%s",
+                          UCS_PP_MAKE_STRING(UCT_TCP_CM_CONN_ACK));
+    }
 }
 
-static ucs_status_t
-uct_tcp_cm_conn_pkt_check_event(const uct_tcp_ep_t *ep,
-                                uct_tcp_cm_conn_event_t expected_event,
-                                uct_tcp_cm_conn_event_t actual_event)
+/* `fmt_str` parameter has to contain "%s" to write event type */
+static void uct_tcp_cm_trace_conn_pkt(const uct_tcp_ep_t *ep,
+                                      const char *fmt_str,
+                                      uct_tcp_cm_conn_event_t event)
 {
+    ucs_log_level_t log_level = UCS_LOG_LEVEL_DEBUG;
+    char event_str[64]        = { 0 };
     char str_addr[UCS_SOCKADDR_STRING_LEN];
+    char msg[128];
 
-    if (expected_event != actual_event) {
-        ucs_error("tcp_ep %p: received wrong CM event (expected: %u, "
-                  "actual: %u) from the peer with iface listener "
-                  "address: %s", ep, expected_event, actual_event,
-                  ucs_sockaddr_str((const struct sockaddr*)&ep->peer_addr,
-                                   str_addr, UCS_SOCKADDR_STRING_LEN));
-        return UCS_ERR_INVALID_PARAM;
+    uct_tcp_cm_event_str(event, event_str, sizeof(event_str));
```

Comment:
can it be a single function, which prints connections?

Reply:
done